mlflow/tensorflow.py

```diff
@@ -657,26 +657,67 @@ def autolog(every_n_iter=100):
                       "1.12 <= v <= 2.0.0 are supported.")
         return
 
+    @gorilla.patch(tensorflow.estimator.Estimator)
+    def train(self, *args, **kwargs):
```

Comment:
Let's also update the documentation of `autolog` to note that both `train` and `export_saved_model` will start/end runs, and that to have everything autologged under a single run the user should wrap their training logic in a `with mlflow.start_run()`?

Reply:
We should also update https://github.com/mlflow/mlflow/blob/master/examples/tensorflow/tf1/train_predict.py and https://github.com/mlflow/mlflow/blob/master/examples/tensorflow/tf2/train_predict_2.py to follow this practice

Reply:
Sure, I'll do the doc updates in this PR and update the examples in a follow-up.